### PR TITLE
Fix compilation for non-lite version

### PR DESCRIPTION
**Pull request by Hopobcn**
Specify library name and version:  **lib/1.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Please consider these changes.
I've tried with v3.9.1, v3.11.4, v3.12.4 with default options and seems to work in my machine (macos)

/cc @uilianries

### CODE REVIEW

File: recipes/protobuf/all/conanfile.py [hunk 1]
```diff
@@ -70,34 +70,65 @@ def _configure_cmake(self):
...
         return self._cmake
 
-    def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+    def _patch_sources(self):
+        if "patches" in self.conan_data:
+            if self.version in self.conan_data["patches"]:
+                for patch in self.conan_data.get("patches", {}).get(self.version, []):
+                    tools.patch(**patch)
```
Comment (uilianries) [inline recipes/protobuf/all/conanfile.py:77]:
```suggestion
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```

File: recipes/protobuf/all/test_package/conanfile.py [hunk 2]
```diff
@@ -9,8 +9,7 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
-        cmake.definitions["protobuf_VERBOSE"] = True
-        cmake.definitions["protobuf_MODULE_COMPATIBLE"] = True
+        #cmake.definitions["protobuf_LITE"] = self.options["protobuf"].lite
```
Comment (uilianries) [inline recipes/protobuf/all/test_package/conanfile.py:12]:
```suggestion
```

Remove if not used

**Review by uilianries — CHANGES_REQUESTED** [review]

**Comment by uilianries** [comment]
@Hopobcn Great job, could you please solve the conflict.

**Comment by Hopobcn** [comment]
I've resolved the conflicts. Since now bincrafters:protobuf/3.12.3 builds, my MR has been reduced to only cosmetic changes. Feel free to cherry-pick or not.

**Review by uilianries — APPROVED** [review]

**Comment by uilianries** [comment]
@Hopobcn Thanks, Your test package looks better!

**Pull request merged**